supabase/migrations/20230611230132_song_instrument.sql

```diff
@@ -0,0 +1,18 @@
+
+create policy "Enable delete for all authenticated users"
+on "public"."song_instrument"
+as permissive
+for delete
+to authenticated
+using (true);
+
+
+create policy "Enable update for all authenticated users"
+on "public"."song_instrument"
+as permissive
+for update
+to authenticated
+using (true);
```

Comment:
Why to all authenticated users?

Reply:
Isn't that what we want? same policies are used for other tables ex song: 
![image](https://github.com/ASVGay/the-rhapsodies/assets/4838158/897df0ab-57a4-432c-a945-ba1822a51f6a)

Reply:
Oh that should not be the case. I should not be able to delete your song and/or song instrument

Reply:
You only should  be able to delete the song instrument if it is your song instrument

Reply:
> Isn't that what we want? same policies are used for other tables ex song: ![image](https://user-images.githubusercontent.com/4838158/245089858-897df0ab-57a4-432c-a945-ba1822a51f6a.png)

Just created a PR to fix this
* #281

Reply:
Ok, i will add a check to the song_instruments delete and update.

In my opinion it's better for PR's to focus on what is added functionally to a PR than explicitly what the PR says what lines are changed. It demotivated developers to move files and functions to more reasonable places. The feedback given here is valid and makes the application more secure, but a lot of the feedback from this and previous branch are semantics that the browser doesn't care about, the tests either.

Reply:
I understand your point and will try to have more empathy because I can see how it can be demotivating to constantly have to fix such little changes. 

However, I don't think that focusing only on the functionality is the way to go since clean code is in my opinion also important. Clean code might not have a lot of impact on the performance of the tests or website, but does have an impact on the productivity of the developer. It is also important to me since this project won't be archived after this sprint, but actively worked on.

Also mentioned with several comments that they are rather optional & can or could implement them. So if you rather would not implement those feedback comments, that is okay as well. :) One of the other reasons I'm giving feedback on such things because I've had it happen to me multiple times when I did not even know about how it could be written in another way but have since stayed with me.

But once again, I do understand where you're coming from and will try to be more empathetic!

Reply:
I agree that clean code is important. The main reason that i do implement these changes you note as optional is that I understand that this project will continue after the school's sprints; and i want to accommodate to that. But i can't mind-read when these new semantics are introduced. For example:
![image](https://github.com/ASVGay/the-rhapsodies/assets/4838158/338319ef-f38b-4582-8bcd-63a052b68f4c)
These toast are not written in a one-line method. In the history of the progress-bar https://github.com/ASVGay/the-rhapsodies/commit/0d6aa1a0a40de66aedb513ab8b89b08da2da5ba4, these also weren't arrow functions.

I did adjust those to arrow functions when I saw that it didn't fit mostly used conventions in the project. These conventions aren't written down, and vary from developer, group and company what they define as clean code.

Reply:
Very valid point!